server/aesdsocket.c

```diff
@@ -0,0 +1,602 @@
+/*
+This file contains the code for server design.
+
+Reference: https://beej.us/guide/bgnet/html/ 
+*/
+
+// ToDo;
+// Try to make both buffers local anf free them
+
+
+#include <stdio.h>
+#include <string.h>
+#include <syslog.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <netdb.h>
+#include <stdlib.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <linux/fs.h>
+#include <sys/stat.h>
+
+#define DEBUG 0
+#define ASCII_NEWLINE 10
+#define PORT_NUMBER 9000
+
+char socket_file_path[50] = "/var/tmp/aesdsocketdata";
+
+int server_socket_fd;
+int client_socket_fd;
+int socket_file_fd;
+char* rx_storage_buffer = NULL;
+
+void sig_handler(int signum)
+{
+    int status;
+
+    if (signum == SIGINT)
+    {
+        printf("\nCaught signal SIGINT, exiting\n");
+        // Syslog the error into the syslog file in /var/log
+        syslog(LOG_DEBUG, "Caught signal SIGINT, exiting");
+    }
+    else if (signum == SIGTERM)
+    {
+        printf("\nCaught signal SIGTERM, exiting\n");
+        // Syslog the error into the syslog file in /var/log
+        syslog(LOG_DEBUG, "Caught signal SIGTERM, exiting");
+    }
+
+    // Freeing the rx_storage_buffer
+    free(rx_storage_buffer);
```

Comment:
@nihalthirunakarasu are all of these functions safe to call in a signal handler?